### PR TITLE
Add worker scheduler to handle the scheduling of the worker task.

### PR DESCRIPTION
**Pull request by nakhbari**
This scheduler is useful because it can also handle cron jobs and can be expanded in the future to meet the needs of our worker's timing.

### CODE REVIEW

File: workers/worker_scheduler.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+require "rufus-scheduler"
+
+module FastlaneCI
+  # Class that handles the scheduling for fastlane.ci workers
+  class WorkerScheduler
+    # Sleep in seconds
+    attr_accessor :sleep_interval
+    # Ex. '5 0 * * *' do something every day, five minutes after midnight
+    # (see "man 5 crontab" in your terminal)
```
Comment (KrauseFx) [inline workers/worker_scheduler.rb]: Nice comment 👍

File: workers/worker_scheduler.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        raise "Either a cron_time or a sleep_interval is mandatory."
+      end
+    end
+
+    def schedule(&block)
+      if !self.sleep_interval.nil?
+        block.call
+        Kernel.sleep(self.sleep_interval)
+      elsif !self.cron_time.nil?
+        self.scheduler.cron(self.cron_time) { block.call }
```
Comment (nakhbari) [inline workers/worker_scheduler.rb]: i have this being checked if the cron job has already been schedule in another CL

File: workers/refresh_config_data_sources_worker.rb [hunk 3]
```diff
@@ -8,8 +9,8 @@ def work
       FastlaneCI::Services.project_service.refresh_repo
     end
 
-    def sleep_interval
-      15
+    def scheduler
+      WorkerScheduler.new(sleep_interval: 15)
```
Comment (nakhbari) [inline workers/refresh_config_data_sources_worker.rb]: I have these being declared on init in another CL

File: workers/worker_scheduler.rb (hunk 2 above)
Comment (taquitos) [reply workers/worker_scheduler.rb]: ❤️

File: workers/refresh_config_data_sources_worker.rb (hunk 3 above)
Comment (taquitos) [reply workers/refresh_config_data_sources_worker.rb]: ❤️

File: workers/worker_scheduler.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+require "rufus-scheduler"
+
+module FastlaneCI
+  # Class that handles the scheduling for fastlane.ci workers
+  class WorkerScheduler
+    # Sleep in seconds
+    attr_accessor :sleep_interval
+    # Ex. '5 0 * * *' do something every day, five minutes after midnight
+    # (see "man 5 crontab" in your terminal)
+    attr_accessor :cron_time
```
Comment (taquitos) [inline workers/worker_scheduler.rb]: I think `cron_schedule` might be a little more descriptive as a name

File: workers/worker_scheduler.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    attr_accessor :scheduler
+
+    def initialize(sleep_interval: nil, cron_time: nil)
```
Comment (taquitos) [inline workers/worker_scheduler.rb]: ❤️ that we can schedule it two ways!

File: workers/worker_scheduler.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      self.sleep_interval = sleep_interval
+      self.cron_time = cron_time
+      self.scheduler = Rufus::Scheduler.new
+
+      if self.sleep_interval.nil? && self.cron_time.nil?
```
Comment (taquitos) [inline workers/worker_scheduler.rb]: Ohhh maybe check to make sure we don't have both `sleep_interval` AND `cron_time` set?

File: workers/worker_scheduler.rb (hunk 4 above)
Comment (nakhbari) [reply workers/worker_scheduler.rb]: done.

File: workers/worker_scheduler.rb (hunk 6 above)
Comment (nakhbari) [reply workers/worker_scheduler.rb]: done.

**Pull request merged**